src/toxcore/onion/request_queue.rs

```diff
@@ -50,4 +50,16 @@ impl<T> RequestQueue<T> { // TODO: unify with DHT?
             clock_elapsed(time) <= timeout
         );
     }
+
+    /// Returns the Values of ping_map.
+    pub fn get_values(&self) -> impl Iterator<Item=(Instant, &T)> {
+        self.ping_map
+            .values()
```

Comment:
Namsoo, add `.filter(|(ping_time, _request_data)| clock_elapsed(*ping_time) <= request_queue.get_timeout())` here, and add mark to doc comment that we return fresh values

Reply:
done